src/control/server/mgmt_svc_test.go

```diff
@@ -130,3 +141,174 @@ func TestCheckMgmtSvcReplica(t *testing.T) {
 		}
 	}
 }
+
+// mockDrpcClient is a mock of the DomainSocketClient interface
+type mockDrpcClient struct {
+	sync.Mutex
+	ConnectOutputError    error
+	CloseOutputError      error
+	CloseCallCount        int
+	SendMsgInputCall      *drpc.Call
+	SendMsgOutputResponse *drpc.Response
+	SendMsgOutputError    error
+}
+
+func (c *mockDrpcClient) IsConnected() bool {
+	return false
+}
+
+func (c *mockDrpcClient) Connect() error {
+	return c.ConnectOutputError
+}
+
+func (c *mockDrpcClient) Close() error {
+	c.CloseCallCount++
+	return c.CloseOutputError
+}
+
+func (c *mockDrpcClient) SendMsg(call *drpc.Call) (*drpc.Response, error) {
+	c.SendMsgInputCall = call
+	return c.SendMsgOutputResponse, c.SendMsgOutputError
+}
+
+func (c *mockDrpcClient) setSendMsgResponse(status drpc.Status, body []byte) {
+	c.SendMsgOutputResponse = &drpc.Response{
+		Status: status,
+		Body:   body,
+	}
+}
+
+func newTestMgmtSvc(log logging.Logger) *mgmtSvc {
+	r := ioserver.NewRunner(log, ioserver.NewConfig())
+
+	var msCfg mgmtSvcClientCfg
+	msCfg.AccessPoints = append(msCfg.AccessPoints, "localhost")
+
+	srv := NewIOServerInstance(log, nil, nil, newMgmtSvcClient(nil, log, msCfg), r)
+	srv.setSuperblock(&Superblock{
+		MS: true,
+	})
+
+	harness := NewIOServerHarness(log)
+	harness.instances = append(harness.instances, srv)
+
+	return newMgmtSvc(harness)
+}
+
+func setupMockDrpcClientBytes(svc *mgmtSvc, respBytes []byte, err error) {
+	mi, _ := svc.harness.GetMSLeaderInstance()
+	client := &mockDrpcClient{}
+	client.setSendMsgResponse(drpc.Status_SUCCESS, respBytes)
+	client.SendMsgOutputError = err
+	mi.setDrpcClient(client)
+}
+
+func setupMockDrpcClient(svc *mgmtSvc, resp proto.Message, err error) {
+	respBytes, _ := proto.Marshal(resp)
+	setupMockDrpcClientBytes(svc, respBytes, err)
+}
+
+func newTestGetACLReq() *mgmtpb.GetACLReq {
+	return &mgmtpb.GetACLReq{
+		Uuid: "testUUID",
+	}
+}
+
+func TestPoolGetACL_NoMS(t *testing.T) {
+	log, buf := logging.NewTestLogger(t.Name())
+	defer common.ShowBufferOnFailure(t, buf)()
+
+	svc := newMgmtSvc(NewIOServerHarness(log))
+
+	resp, err := svc.PoolGetACL(nil, newTestGetACLReq())
+
+	if resp != nil {
+		t.Errorf("Expected no response, got: %+v", resp)
+	}
+
+	if err == nil {
+		t.Fatal("Expected an error, got nil")
+	}
+
+	if !strings.Contains(err.Error(), "no managed instances") {
```

Comment:
I added a `cmpErr(t, want, got)` test helper recently. It's defined in the server package, but we might want to move it to the common package. Could simplify this check.